front/src/modules/timesStops/hooks/useOutputTableData.tsx

```diff
@@ -0,0 +1,111 @@
+import { useMemo } from 'react';
+
+import { keyBy } from 'lodash';
+import { useSelector } from 'react-redux';
+
+import type {
+  PathPropertiesFormatted,
+  SimulationResponseSuccess,
+} from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { formatSuggestedOperationalPoints } from 'modules/pathfinding/utils';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+import { getSelectedTrainId } from 'reducers/osrdsimulation/selectors';
+import { ISO8601Duration2sec, datetime2sec, secToHoursString } from 'utils/timeManipulation';
+
+import useComputedMargins from './useComputedMargins';
+import { findNextScheduledPoint } from '../utils';
+
+function useOutputTableData(
+  simulatedTrain: SimulationResponseSuccess,
+  pathProperties: PathPropertiesFormatted,
+  operationalPoints: OperationalPointWithTimeAndSpeed[],
+  startTime: string
+) {
+  const selectedTrainId = useSelector(getSelectedTrainId);
+
+  const { data: selectedTrainSchedule = {} as TrainScheduleResult } =
+    osrdEditoastApi.endpoints.getV2TrainScheduleById.useQuery(
+      {
+        id: selectedTrainId as number,
+      },
+      { skip: !selectedTrainId }
+    );
+
+  const trainScheduleByAt = keyBy(selectedTrainSchedule.schedule, 'at');
+  const suggestedOperationalPoints: SuggestedOP[] = useMemo(
+    () =>
+      formatSuggestedOperationalPoints(
+        pathProperties.operationalPoints,
+        pathProperties.geometry,
+        100
+      ),
+    [pathProperties.operationalPoints, pathProperties.geometry]
+  );
+
+  const pathPointsWithOpPointIndices = selectedTrainSchedule.path.map((pathPoint) => {
+    const correspondingOpPointIndex = suggestedOperationalPoints.findIndex((sugOpPoint) => {
+      if ('uic' in pathPoint) {
+        return sugOpPoint.uic === pathPoint.uic && sugOpPoint.ch === pathPoint.secondary_code;
+      }
+      return false;
+    });
+    return {
+      ...pathPoint,
+      correspondingOpPointIndex,
+    };
+  });
+  const trainSchedulePathByUic = keyBy(pathPointsWithOpPointIndices, (pathPoint) => {
+    if ('uic' in pathPoint) {
+      return `${pathPoint.uic}-${pathPoint.secondary_code}`;
+    }
+    if ('operational_point' in pathPoint) {
+      return pathPoint.operational_point;
+    }
+    return '';
+  });
```

Comment:
You can filter the pathSteps without uic above (in pathPointsWithOpPointIndices) and then use this instead:
```suggestion
  const trainSchedulePathByUic = keyBy(pathStepsWithOpPointIndices, (pathStep) =>
    'uic' in pathStep ? `${pathStep.uic}-${pathStep.secondary_code}` : ''
  );
   );
```

Reply:
done